.github/workflows/release.yml

```diff
@@ -1,59 +1,174 @@
-name: release
+name: Build and Release
+
 on:
+  workflow_dispatch:
+  release:
+    types: [published]
   push:
-    tags:
-      - v*
+    branches:
```

Comment:
what's the goal of building binaries on master and PRs?
be aware of build failures?

Reply:
yes

Reply:
But I don't know much about golang, so I don't know if it is necessary to test the binary file build. What do you think?

Reply:
it depends on the tests coverage: ideally if tests run then it should also compile

@stefanbenten what do you think?

Reply:
I do not think we need to build the assets for PRs, since the tests should cover the actual compile yeah.
For master its debatable if users would like to get the binaries for non-releases, generally should not be a big deal tho.

Reply:
@wc7086 please remove then, thanks